tests/e2e/test_e2e_internal_trades.py

```diff
@@ -0,0 +1,161 @@
+            a for a in internal_token_transfers if a.transfer_type == 'INTERNAL_TRADE']
+        self.assertEqual(len(internal_buffer_trades), 0*2)
+
+    def test_buffer_trade_with_missing_price_from_pricesUSD(self):
+        '''0x80ae1c6a5224da60a1bf188f2101bd154e29ef71d54d136bfd1f6cc529f9d7ef
+            crvcx is not part of the price list
+        '''
+        dune_connection = DuneAnalytics.new_from_environment()
+        internal_token_transfers = get_internal_token_transfers(
+            dune=dune_connection,
+            tx_hash='0x80ae1c6a5224da60a1bf188f2101bd154e29ef71d54d136bfd1f6cc529f9d7ef',
+            period_start=datetime.strptime('2022-03-01', "%Y-%m-%d"),
+            period_end=datetime.strptime('2022-03-12', "%Y-%m-%d"),
+        )
+        internal_buffer_trades = [
+            a for a in internal_token_transfers if a.transfer_type == 'INTERNAL_TRADE']
+        self.assertEqual(len(internal_buffer_trades), 1*2)
+        self.assertEqual(calculate_slippage_for_token(Address(
+            '0xD533a949740bb3306d119CC777fa900bA034cd52'), internal_token_transfers), 0)
+        self.assertEqual(calculate_slippage_for_token(Address(
+            '0x62B9c7356A2Dc64a1969e19C23e4f579F9810Aa7'), internal_token_transfers), 0)
+
+    def test_buffer_trade_without_gp_settlement_price(self):
+        '''0x20ae31d11dba93d372ecf9d0cb387ea446e88572ce2d3d8e3d410871cfe6ec49
+            This settlement has an internal tx between WETH and FOLD, but there is not settlement prices for WETH in the call data
+        '''
+        dune_connection = DuneAnalytics.new_from_environment()
+        internal_token_transfers = get_internal_token_transfers(
+            dune=dune_connection,
+            tx_hash='0x20ae31d11dba93d372ecf9d0cb387ea446e88572ce2d3d8e3d410871cfe6ec49',
+            period_start=datetime.strptime('2022-03-01', "%Y-%m-%d"),
+            period_end=datetime.strptime('2022-03-12', "%Y-%m-%d"),
+        )
+        internal_buffer_trades = [
+            a for a in internal_token_transfers if a.transfer_type == 'INTERNAL_TRADE']
+        self.assertEqual(len(internal_buffer_trades), 1*2)
+
+    def test_it_recognizes_slippage(self):
+        '''0x703474ed43faadc35364254e4f9448e275c7cfe9cf60beddbdd68a462bf7f433
+           Paraswap returns to little ust, but our buffers make up for it
+        '''
+        dune_connection = DuneAnalytics.new_from_environment()
+        internal_token_transfers = get_internal_token_transfers(
+            dune=dune_connection,
+            tx_hash='0x703474ed43faadc35364254e4f9448e275c7cfe9cf60beddbdd68a462bf7f433',
+            period_start=datetime.strptime('2022-03-01', "%Y-%m-%d"),
+            period_end=datetime.strptime('2022-03-12', "%Y-%m-%d"),
+        )
+        internal_buffer_trades = [
+            a for a in internal_token_transfers if a.transfer_type == 'INTERNAL_TRADE']
+        self.assertEqual(len(internal_buffer_trades), 0*2)
+        # We lost 58 ust dollars:
```

Comment:
I am not actually sure what to suggest here, it just looks like a typo. Could be USDT maybe or maybe it was meant to say US dollars.
```suggestion
        # We lost 58 USD
```

Reply:
It's a token. UST